shell/dialog/SloDialog.vue

```diff
@@ -0,0 +1,97 @@
+<script>
+import { Card } from '@components/Card';
+import AsyncButton from '@shell/components/AsyncButton';
+import { IS_SSO, LOGGED_OUT } from '@shell/config/query-params';
+
+export default {
+  components: { Card, AsyncButton },
+
+  props: {
+    authProvider: {
+      type:     Object,
+      required: true
+    }
+  },
+
+  computed: {
+    name() {
+      return this.authProvider?.nameDisplay;
+    }
+  },
+
+  methods: {
+    async doLogout(logoutAll = false) {
+      const options = { force: true, provider: this.authProvider };
+
+      if (logoutAll) {
+        // SLO - Single Log Out
+        options.slo = true;
+        options.provider = this.authProvider;
+
+        await this.$store.dispatch('auth/logout', options, { root: true });
+
+        this.$emit('close');
+      } else {
+        // Rancher Log Out (ensure we show the logging out page as per standard log out)
+        this.$router.replace({ name: 'auth-logout', query: { [LOGGED_OUT]: true, [IS_SSO]: true } });
+      }
+    },
+
+    cancel() {
+      this.$emit('close');
+    }
+  }
+};
+</script>
+
+<template>
+  <Card
+    class="prompt-remove"
+    :show-highlight-border="false"
+  >
+    <h4
+      slot="title"
+      class="text-default-text"
+    >
+      {{ t('promptSlo.title', { name }) }}
+    </h4>
```

Comment:
This slot syntax has been deprecated since Vue 2.6[^1]. We can use a template tag with `v-slot:title` or the shorthand `#title` to avoid regressions after migrating to Vue3. 

[^1]: https://v2.vuejs.org/v2/guide/components-slots.html#Deprecated-Syntax

```suggestion
    <template #title>
      <h4 class="text-default-text">
        {{ t('promptSlo.title', { name }) }}
      </h4>
    </template>
```